packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -20,7 +21,7 @@ abstract class FlutterCommand extends Command {
   FlutterCommandRunner get runner => super.runner;
 
   /// Whether this command needs to be run from the root of a project.
-  bool get requiresProjectRoot => true;
+  bool get requiresProjectRoot => !_targetSpecified;
```

Comment:
@devoncarew Are you saying that each subclass that calls addTargetOption() also needs to override requiresTarget?